stdlib/assoc.mc

```diff
@@ -27,6 +27,13 @@ let assocInsert : AssocTraits k -> k -> v -> AssocMap k v -> AssocMap k v =
                     (lam i. set m i (k,v))
                     (index (lam t. traits.eq k t.0) m)
 
+-- 'assocConstruct traits ls' constructs a new association map from a sequence
```

Comment:
That was my original naming, but I avoided it since we have a convention that says we should not write conversion functions in that form (it should be `type2type`). With the `assoc` namespacing, it would then be `assocseq2assoc`, which looks horrible. I'm starting to agree that we should use the other form instead (I believe it was `typeToType` which I guess would translate to `assocSeqToAssoc` in this case).